### DISCUSSION TITLE
Are there any Updates on the wireless xbox one adapter?

### DISCUSSION THREAD

**Discussion by Sumpftier** [cathery/sys-con#178] [Controller requests]

**Comment by cathery** [comment]
not really

**Comment by sirocyl** [comment]
Xbox One Wireless compatibility is a bit complicated - I looked into it earlier.

https://github.com/medusalix/xow has a good reverse-engineered reimplementation. The major hurdles are:
- Communicating with the MT76 chipset
- Downloading the Microsoft driver package containing the dongle firmware, unpacking it, and sending it to the dongle
- Establishing the packet/socket connection with the Xbox One controller over its proprietary WLAN channel
- Implementing the GIP protocol used by the wireless connection - it's not HID.

A lot of the code there could be portable and usable here, but a lot will have to be rewritten to target away from Linux.

**Comment by phobos85br** [comment]
So, any news yet?
I ake myself available to help with whatever is needed!
I opened an issue but closed it because this feature is being discussed here.
